aptos-move/aptos-e2e-comparison-testing/src/execution.rs

```diff
@@ -191,163 +252,219 @@ impl Execution {
         } else {
             let res = res_main_opt.unwrap();
             if let Ok(res_ok) = res {
-                println!(
+                self.output_result_str(format!(
                     "version:{}\nwrite set:{:?}\n events:{:?}\n",
                     cur_version, res_ok.0, res_ok.1
-                );
+                ));
             } else {
-                println!(
+                self.output_result_str(format!(
                     "execution error {} at version: {}, error",
                     res.unwrap_err(),
                     cur_version
-                );
+                ));
             }
         }
     }
 
     fn execute_code(
         &self,
-        state: &FakeDataStore,
-        features: &Features,
+        version: Version,
+        state: FakeDataStore,
         package_info: &PackageInfo,
         txn: &Transaction,
         compiled_package_cache: &HashMap<PackageInfo, HashMap<ModuleId, Vec<u8>>>,
+        debugger_opt: Option<Arc<dyn AptosValidatorInterface + Send>>,
+        v2_flag: bool,
     ) -> Option<Result<(WriteSet, Vec<ContractEvent>), VMStatus>> {
         let executor = FakeExecutor::no_genesis();
         let mut executor = executor.set_not_parallel();
-        *executor.data_store_mut() = state.clone();
+        *executor.data_store_mut() = state;
         if let Transaction::UserTransaction(signed_trans) = txn {
             let sender = signed_trans.sender();
             let payload = signed_trans.payload();
             if let TransactionPayload::EntryFunction(entry_function) = payload {
-                // always load 0x1 modules
+                // Always load 0x1 modules
                 load_aptos_packages_to_executor(&mut executor, compiled_package_cache);
-                // Load other modules
+                // Load modules
                 if package_info.is_compilable() {
                     load_packages_to_executor(&mut executor, package_info, compiled_package_cache);
                 }
                 let mut senders = vec![sender];
                 senders.extend(TransactionMetadata::new(signed_trans).secondary_signers);
-                return Some(executor.try_exec_entry_with_features(
-                    senders,
-                    entry_function,
-                    features,
-                ));
-            } else if let TransactionPayload::Multisig(multi_sig) = payload {
-                assert!(multi_sig.transaction_payload.is_some());
-                println!("Multisig transaction is not supported yet");
+                let enable_v7 = |features: &mut Features| {
+                    if v2_flag {
+                        features.enable(FeatureFlag::VM_BINARY_FORMAT_V7);
+                    } else {
+                        features.enable(FeatureFlag::VM_BINARY_FORMAT_V6);
+                    }
+                };
+                if let Some(debugger) = debugger_opt {
+                    let data_view =
```

Comment:
You can have `let resolver = if .. else ...; ` at the start, and then straight line code to fetch features, enable v7, and execute to avoid duplication

Reply:
Yeah, I tried it but compilation error: "`if` and `else` have incompatible types" is generated because one branch returns `StorageAdapter<'_, DataStateView>` and the other returns `StorageAdapter<'_, FakeDataStore>`. Is there anything I could do to fix it?

Reply:
Ah right, then don't bother :)